api/logical.go

```diff
@@ -62,26 +62,7 @@ func (c *Logical) ReadWithData(path string, data map[string][]string) (*Secret,
 }
 
 func (c *Logical) ReadWithDataWithContext(ctx context.Context, path string, data map[string][]string) (*Secret, error) {
-	ctx, cancelFunc := c.c.withConfiguredTimeout(ctx)
-	defer cancelFunc()
-
-	r := c.c.NewRequest(http.MethodGet, "/v1/"+path)
-
-	var values url.Values
-	for k, v := range data {
-		if values == nil {
-			values = make(url.Values)
-		}
-		for _, val := range v {
-			values.Add(k, val)
-		}
-	}
-
-	if values != nil {
-		r.Params = values
-	}
-
-	resp, err := c.c.rawRequestWithContext(ctx, r)
+	resp, err := c.ReadRawWithDataWithContext(ctx, path, data)
```

Comment:
this does in fact seem to have broken the cancel-code :/

Reply:
@kitography Updated, turns out deferring the defer into the helper is the wrong way to go, I've fixed that up.